### PR TITLE
[feat]:[PIE-4659]: Migrating Redis cache to enterprise for ng manager and cv

### PR DESCRIPTION
**Pull request by ankittiwari-harness**
## Migrating Redis cache to enterprise for ng manager and cv
1) Adding config changes in ng amnager, sto manager and cv for using enterprise cache manager
2) Migrated sdkEventsCache to Enterprise Redis
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36248)

**Comment by ankittiwari-harness** [comment]
trigger codeformat

**Comment by ankittiwari-harness** [comment]
trigger messagecheck

**Comment by ankittiwari-harness** [comment]
trigger codeformat

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 20 of 22 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 22 file(s) to review. 1 change requests. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`120-ng-manager/container/scripts/replace_configs.sh` line 401 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96UTa-AvVlz6rGd7BM:-N96UTa-AvVlz6rGd7BN:b-vtms65) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/120-ng-manager/container/scripts/replace_configs.sh#L401)):*
> ```Shell
> 
> if [[ "$REDIS_SCRIPT_CACHE" == "false" ]]; then
>   yq write -i $ENTERPRISE_REDISSON_CACHE_FILE useScriptCache false
> ```

we need to make changes in build/container/scripts/replace_configs.sh as well
___
*[`300-cv-nextgen/cv-nextgen-config.yml` line 160 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96SC0A2EXTxm7NADOt:-N96SC0A2EXTxm7NADOu:b-818w1y) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/300-cv-nextgen/cv-nextgen-config.yml#L160)):*
> ```YAML
>   cacheNamespace: ""
>   disabledCaches: [ ]
>   enterpriseCacheEnabled: true
> ```

build_dist.sh, Build.bazel, replace_configs.sh changes for cv are not there
___
*[`315-sto-manager/sto-manager-config.yml` line 235 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96SU88AjP59iwr9S9X:-N96SU88AjP59iwr9S9Y:byppj8) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/315-sto-manager/sto-manager-config.yml#L235)):*
> ```YAML
>   corePoolSize: ${PLAN_CREATOR_SERVICE_EXECUTOR_POOL_CORE_SIZE:-2}
>   maxPoolSize: ${PLAN_CREATOR_SERVICE_EXECUTOR_POOL_MAX_SIZE:-3}
>   idleTime: ${PLAN_CREATOR_SERVICE_EXECUTOR_POOL_IDLE_TIME:-30}
> ```

BUILD.bazel file changes for sto are not there
___
*[`dockerization/cv-nextgen/Dockerfile-cv-nextgen-cie-jdk` line 35 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96TotuCXc29ih8LCTS:-N96TotuCXc29ih8LCTT:b-ysha8z) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/dockerization/cv-nextgen/Dockerfile-cv-nextgen-cie-jdk#L35)):*
> ```
> 
> COPY --chown=65534:65534 inject-saas-apm-bins-into-dockerimage.sh /opt/harness
> RUN /opt/harness/inject-saas-apm-bins-into-dockerimage.sh && rm -rf /opt/harness/inject-saas-apm-bins-into-dockerimage.sh
> ```

dockerisation folder changes for ci are not there
___
*[`scripts/jenkins/portal-openjdk-bazel.sh` line 199 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96T7pf7URR_1QybwkJ:-N96T7pf7URR_1QybwkK:b-vnlmer) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/scripts/jenkins/portal-openjdk-bazel.sh#L199)):*
> ```Shell
> cp ../../332-ci-manager/config/key.pem .
> cp ../../332-ci-manager/config/cert.pem .
> cp ../../332-ci-manager/service/src/main/resources/redisson-jcache.yaml .
> ```

enterprise redisson file not added for ci, plus changes for ci service not done

**Comment by sahilhindwani-harness** [comment]
*[`120-ng-manager/config.yml` line 427 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96UsMiBLOhx4cTkZsr:-N96UsMiBLOhx4cTkZss:b-d4n8rg) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/120-ng-manager/config.yml#L427)):*
> ```YAML
>   cacheNamespace: ""
>   disabledCaches: [ ]
>   enterpriseCacheEnabled: true
> ```

${ENTERPRISE_CACHE_ENABLED:-false}

nit: We can use above so that we don't need to add this in replace_config.sh



_Code quote:_
```YAML
true
```

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 20 of 22 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 2 change requests. 0 of 2 approvals obtained. Waiting on team(s): sto.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 8 file(s) to review. 2 change requests. 0 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`120-ng-manager/container/scripts/replace_configs.sh` line 401 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96UTa-AvVlz6rGd7BM:-N9B8D8n07eKcLphkNr5:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/120-ng-manager/container/scripts/replace_configs.sh#L401)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

we need to make changes in build/container/scripts/replace_configs.sh as well
</blockquote></details>

done
___
*[`300-cv-nextgen/cv-nextgen-config.yml` line 160 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96SC0A2EXTxm7NADOt:-N9B9JpA0S-dHsy5X3HE:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/300-cv-nextgen/cv-nextgen-config.yml#L160)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

build_dist.sh, Build.bazel, replace_configs.sh changes for cv are not there
</blockquote></details>

done
___
*[`315-sto-manager/sto-manager-config.yml` line 235 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96SU88AjP59iwr9S9X:-N9B9_DW2vr5biAVk_8E:bi6l9uj) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/315-sto-manager/sto-manager-config.yml#L235)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

BUILD.bazel file changes for sto are not there
</blockquote></details>

@ankittiwari-harness this  example "//120-ng-manager/src/main/resources/enterprise-redisson-jcache.yaml", i think is still left
___
*[`dockerization/cv-nextgen/Dockerfile-cv-nextgen-cie-jdk` line 35 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96TotuCXc29ih8LCTS:-N9B9jJ717PmfXxFb9af:b-ekmyve) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/dockerization/cv-nextgen/Dockerfile-cv-nextgen-cie-jdk#L35)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

dockerisation folder changes for ci are not there
</blockquote></details>

this is i think is also pending
___
*[`scripts/jenkins/portal-openjdk-bazel.sh` line 199 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96T7pf7URR_1QybwkJ:-N9B9lY6AcjqZ85Wd1yQ:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/scripts/jenkins/portal-openjdk-bazel.sh#L199)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

enterprise redisson file not added for ci, plus changes for ci service not done
</blockquote></details>

done

**Review by archit-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 16 file(s) to review. 1 change requests. 1 of 2 approvals obtained. Waiting on team(s): sto. 

___
*[`315-sto-manager/sto-manager-config.yml` line 235 at r3](https://reviewable.io/reviews/harness/harness-core/36248#-N96SU88AjP59iwr9S9X:-N9BpcVNBCDLPeXoeJc6:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/397252369a6bcc4114d41235a3ce4b090c0fb0d7/315-sto-manager/sto-manager-config.yml#L235)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@ankittiwari-harness this  example "//120-ng-manager/src/main/resources/enterprise-redisson-jcache.yaml", i think is still left
</blockquote></details>

done

**Review by AbhijithHarness — DISMISSED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 16 file(s) to review. 1 change requests. Waiting on team(s): sto.

**Review by shubham-in10se — DISMISSED** [review]
> Reviewed 19 of 22 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, 8 of 8 files at r4, 8 of 8 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 1 change requests. Waiting on team(s): sto.

**Comment by ankittiwari-harness** [comment]
> > Reviewed 20 of 22 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)_ status: 2 change requests. 0 of 2 approvals obtained. Waiting on team(s): sto.

 The config related improvements has to be done across multiple projects. Can we pick it up as part of a broader refactoring.

**Review by sahilhindwani-harness — DISMISSED** [review]
> Reviewed 8 of 8 files at r4, 8 of 8 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: Waiting on team(s): sto.

**Comment by ankittiwari-harness** [comment]
trigger ti4

File: 120-ng-manager/container/BUILD.bazel [hunk 1]
```diff
@@ -21,6 +21,7 @@ pkg_tar(
     srcs = [
         "//120-ng-manager:config.yml",
         "//120-ng-manager:keystore.jks",
+        "//120-ng-manager/src/main/resources/enterprise-redisson-jcache.yaml",
```
Comment (kshitij-harness) [inline 120-ng-manager/container/BUILD.bazel:24]: Shouldn't this be resources:enterprise-redisson-jcache.yaml

File: 332-ci-manager/build/container/BUILD.bazel [hunk 2]
```diff
@@ -21,6 +21,7 @@ pkg_tar(
     srcs = [
         "//332-ci-manager/config:ci-manager-config.yml",
         "//332-ci-manager/config:keystore.jks",
+        "//332-ci-manager/service/src/main/resources/enterprise-redisson-jcache.yaml",
```
Comment (kshitij-harness) [inline 332-ci-manager/build/container/BUILD.bazel:24]: Shouldn't this be resources:enterprise-redisson-jcache.yaml

File: 332-ci-manager/container/BUILD.bazel [hunk 3]
```diff
@@ -21,6 +21,7 @@ pkg_tar(
     srcs = [
         "//332-ci-manager/config:ci-manager-config.yml",
         "//332-ci-manager/config:keystore.jks",
+        "//332-ci-manager/service/src/main/resources/enterprise-redisson-jcache.yaml",
```
Comment (kshitij-harness) [inline 332-ci-manager/container/BUILD.bazel:24]: Shouldn't this be resources:enterprise-redisson-jcache.yaml

File: 332-ci-manager/service/src/main/resources/META-INF/services/enterprise-redisson-jcache.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+singleServerConfig : address : "redis://127.0.0.1:6379" codec
```
Comment (kshitij-harness) [inline 332-ci-manager/service/src/main/resources/META-INF/services/enterprise-redisson-jcache.yaml:1]: Could you format this file for better readability?

**Review by kshitij-harness — COMMENTED** [review]
Are the changes tested in pre QA environment?
If yes, kindly add that in description.

File: 332-ci-manager/container/BUILD.bazel (hunk 3 above)
Comment (ankittiwari-harness) [reply 332-ci-manager/container/BUILD.bazel]: done in revision

File: 332-ci-manager/service/src/main/resources/META-INF/services/enterprise-redisson-jcache.yaml (hunk 4 above)
Comment (ankittiwari-harness) [reply 332-ci-manager/service/src/main/resources/META-INF/services/enterprise-redisson-jcache.yaml]: done in revision

**Comment by ankittiwari-harness** [comment]
> Are the changes tested in pre QA environment? If yes, kindly add that in description.

Changes will be tested in Pre QA once merged in develop.

**Review by kshitij-harness — APPROVED** [review]
> Reviewed 16 of 22 files at r1, 1 of 1 files at r3, 2 of 8 files at r4, 7 of 8 files at r5, 5 of 6 files at r6, 2 of 2 files at r7, 5 of 5 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: 12 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): sto.

**Review by archit-harness — APPROVED** [review]
> Reviewed 5 of 6 files at r6, 2 of 2 files at r7, 5 of 5 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36248)* status: Waiting on team(s): sto.

**Comment by ankittiwari-harness** [comment]
trigger ti0

**Pull request merged**